src/dhtproto/client/internal/NodeHashRanges.d

```diff
@@ -57,6 +59,115 @@ public final class NodeHashRanges : NodeHashRangesBase
 
     /***************************************************************************
 
+        Helper encapsulating the node-selection logic required by requests that
+        get a single record from the DHT. Namely:
+
+            During a data redistribution, more than one node may be responsible
+            for a given key. In this case, the node that was most recently
+            reported as being responsible for the key is queried first, followed
+            by others (in order) until the record is located, an error occurs,
+            or no node has the record.
+
+        TODO: test the logic for retrying the request on other nodes which
+        previously covered the hash. This will require a full neo implementation
+        of the Redistribute request. See
+        https://github.com/sociomantic/dhtnode/issues/624
```

Comment:
non-tsunami link.

Reply:
:+1: I'll create a new issue in tsunami.